### ISSUE TITLE
wrong management API used in k8s for liveness probes

### ISSUE THREAD

**Issue by xetys** [jhipster/generator-jhipster#10770]
##### **Overview of the issue**

When generating k8s manifests using `jhipster kubernetes`, the generator sets `/managment/info` for the liveness probe. This endpoint is only visible for authenticated users, so the health check fails and k8s is restarting the pod all the time due to liveness probe fail.

##### **Reproduce the error**

Generate k8s manifests for application of any type. This will generate a block in `deployment.yaml` like this:

``` yaml
        livenessProbe:
          httpGet:
            path: /management/info
            port: http
          initialDelaySeconds: 120
```

which leads to a failing probe

##### **Suggest a Fix**

Using `/management/health` for both, readiness and liveness probe

##### **JHipster Version(s)**

6.4.X and current master

-   [x] Checking this box is mandatory (this is just to show you read everything)

**Comment by atomfrede** [comment]
I think it was changed on purpose as the liveness probe should return if the application is running. Using the health endpoint for the liveness probe would restart the pod even if e.g. there is only a temporary connection failure to the database (which should only remove the pod from the load balancer). But in case the `/info` endpoint is never reachable thats of course also not good.

See https://github.com/jhipster/generator-jhipster/issues/10547

**Comment by xetys** [comment]
I found it just right now... So I get the point about switching to info, but the way how it is now is a service crashing all 2 minutes. On a quite low cluster, it also takes 45secs to boot up...so it is one minute up, one down...if you count eureka service discovery propagation time...the services just will be down all the time :grimacing:

**Comment by atomfrede** [comment]
:) Yes thats not correct right now obviously and not the intention. Can we make the info endpoint to be availble to not authenticated users and hide sensitive information?

**Comment by xetys** [comment]
I tried that out, and you can't configure the info endpoint the same way as health.

Anyways, what we need is a sign of life. Therefore, a TCP socket test should be enough, by defining:

```yaml
livenessProbe:
            tcpSocket:
              port: http
```

**Comment by PierreBesson** [comment]
@xetys I am surprised it is so slow for you to start your apps. Also, the /info endpoint should be publicly available for all users just with different information inside. Are you using an older version of spring boot for which it is not the case ?

Anyway, my plan would be to create a new /readiness and /liveness endpoints as default in JHipster as soon as we migrate to spring boot 2.2 and can use Health indicator groups which will make it nicer and allow customization by the user.
See: https://docs.spring.io/spring-boot/docs/current/reference/html/production-ready-features.html#health-groups
My plan was to use the /info endpoint only temporarily because using the same endpoint for readiness and liveness probe is really bad and can cause cascading failures.

**Comment by xetys** [comment]
don't worry about the startup speed...I'm testing on a very small cluster...while on the other I use eureka and this makes startup time with sync time longer....but anyways...

I was testing it with the current JHI version, and therefore spring boot 2.X. There is a definition for
```
  endpoint:
    health:
      show-details: when-authorized
      roles: 'ROLE_ADMIN'
```
because the `HealthEndpointProperties` allows to define this. There is no similar configuration for the Info endpoint.

So we should build our own health endpoints, or use TCP sockets...

**Comment by PierreBesson** [comment]
It seems that you are right ! However I can't explain why it worked for me.

**Comment by FlorinAlexandru** [comment]
I saw the same issue when containerizing an application. What is, if any, the recommended workaround?

**Comment by PierreBesson** [comment]
@FlorinAlexandru The workaround would be to remove the liveness check.

**Comment by xetys** [comment]
OK, here is my proposal:

let's make a PR which sets the liveness probe back to health, until we merge Spring Boot 2.2. Then we can do another PR with the ehanced logic @PierreBesson described.

WDYT?

**Comment by xetys** [comment]
btw, the fix is part of #10763 

so waiting until that PR is checked and hopefully merged

**Comment by PierreBesson** [comment]
I would rather remove the liveness check entirely for now.

**Comment by FlorinAlexandru** [comment]
I saw the problem but I don't understand it very well. Can someone explain it to me please?

I created an app using jhipster command line tool. Afterwards, I used the sub-command jhispter kubernetes to  generate the k8s files. I deployed then the app in AWS K8s. I logged to a pod, installed wget and run some wget command against _/management/info_  and _/management/health_. The output is bellow: 

`root@demo-6549b6f8c-ww9sk:/home# wget localhost:8080/management/health
--2019-12-02 09:52:49--  http://localhost:8080/management/health
Resolving localhost (localhost)... 127.0.0.1, ::1
Connecting to localhost (localhost)|127.0.0.1|:8080... connected.
HTTP request sent, awaiting response... 200 OK
Length: unspecified [application/vnd.spring-boot.actuator.v2+json]
Saving to: ‘health’

health                                                          [ <=>                                                                                                                                     ]      15  --.-KB/s    in 0s

2019-12-02 09:52:49 (1.32 MB/s) - ‘health’ saved [15]

root@demo-6549b6f8c-ww9sk:/home# wget localhost:8080/management/info
--2019-12-02 09:52:53--  http://localhost:8080/management/info
Resolving localhost (localhost)... 127.0.0.1, ::1
Connecting to localhost (localhost)|127.0.0.1|:8080... connected.
HTTP request sent, awaiting response... 200 OK
Length: unspecified [application/vnd.spring-boot.actuator.v2+json]
Saving to: ‘info’

info                                                            [ <=>                                                                                                                                     ]     199  --.-KB/s    in 0s

2019-12-02 09:52:53 (3.55 MB/s) - ‘info’ saved [199]

root@demo-6549b6f8c-ww9sk:/home# more health
{"status":"UP"}
root@demo-6549b6f8c-ww9sk:/home# more  info
{"display-ribbon-on-profiles":"dev","build":{"artifact":"demogw","name":"Demogw","time":"2019-11-15T13:44:50.534Z","version":"0.0.1-SNAPSHOT","group":"com.mycompany.myapp"},"activeProfiles":["prod"]}
root@demo-6549b6f8c-ww9sk:/home#`

For the /management/info I receive 200 status code. Based on K8s [docs](https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#define-a-liveness-http-request), any code that validates the condition 200<= STATUS_CODE  
< 400 it should indicate a success. I expect that k8s receives the same output as I did when calling /management/info. What am I missing?

**Comment by PierreBesson** [comment]
@FlorinAlexandru please use stackoverflow for general questions.

**Comment by xetys** [comment]
@PierreBesson as #10763 was merged, the endpoints are currently back to health.

As we discussed this here, I would suggest to keep it like this, until Spring Boot 2.2 is merged. Then I can PR a better solution with Health Endpoint customization

**Comment by xetys** [comment]
P.S. I reopen this, to keep in mind this is still to do

**Comment by pascalgrimaud** [comment]
thanks @xetys

**Comment by hanoisteve** [comment]
So how would I create my own url for liveness and readiness probe that does not require authentication? Right now I am getting connection refused: I will add the error here since the specific error message was not mentioned above doing a describe on the deployment I see:
 Readiness probe failed: Get http://192.xyz.yyy.155:8080/management/health: dial tcp 192.xyz.yyy.155:8080: connect: connection refused

and I am using the following deployment:
          readinessProbe:
            httpGet:
              path: /management/health
              port: http
            initialDelaySeconds: 20
            periodSeconds: 15
            failureThreshold: 6
          livenessProbe:
            httpGet:
              path: /management/health
              port: http
            initialDelaySeconds: 120
      imagePullSecrets:

For those of us who are willing to write our own service for readiness how do we do that? How to exclude from the requirement to authenticate? What is solution? Use kubernetes secret? Normally if authentication is required to.a system in kubernetes you just declare a secret maybe the same should be possible for livenessprobe url?

**Comment by pascalgrimaud** [comment]
@xetys : FYI, the Spring Boot 2.2 branch has been merged

**Comment by xetys** [comment]
thx!

**Comment by pascalgrimaud** [comment]
any progress / interest on this ?
who want to contribute ?

**Comment by atomfrede** [comment]
I think with boot 2.3 and the [dedicated kubernetes probes](https://docs.spring.io/spring-boot/docs/2.3.0.M4/reference/html/production-ready-features.html#production-ready-kubernetes-probes) this will be "fixed"?

**Comment by AlexandreCassagne** [comment]
@atomfrede Yes! I am testing on my own deployment, happy to provide feedback once it's "live" :-)

**Comment by DanielFran** [comment]
Closed with #11871
